test/methods.test.js

```diff
@@ -760,3 +760,43 @@ test('register all known HTTP methods', t => {
   t.ok(findMyWay.find('M-SEARCH', '/test'))
   t.equal(findMyWay.find('M-SEARCH', '/test').handler, handlers['M-SEARCH'])
 })
+
+test('off removes all routes without checking constraints if no constraints are specified', t => {
+  t.plan(1)
+
+  const findMyWay = FindMyWay()
+
+  findMyWay.on('GET', '/test', {}, (req, res) => {})
+  findMyWay.on('GET', '/test', { constraints: { host: 'example.com' } }, (req, res) => {})
+
+  findMyWay.off('GET', '/test')
```

Comment:
If I want to remove the first `/test` route without constraints, 
will the following code work?

```
findMyWay.off('GET', '/test', {})
```

Reply:
No but you could delete both route with 
```js
findMyWay.off('GET', '/test')
``` 
But you are right that there is currently no way to delete only the route without constraint. But that should be possible. But then it is no longer possible to delete multiple routes that differ only in the constraint with one call. One possibility would be to introduce an option like `deleteAll`, which would then take care of deleting all routes that differ only by the constraint.

Reply:
@matthyk If the absence of an argument will delete all routes, and an empty object will only delete a route without constraints, will it be convenient?

Reply:
@ivan-tymoshenko Absolutely not 😄 I'll look at it again tomorrow morning

Reply:
@matthyk why not? It sounds pretty logical to me. If you pass the constraint argument, you want to delete only the route that match the constrains. Route with no constraints will match with empty object. If you want to delete the route with all it's constrained variations you will skip the constraint argument.

Reply:
From my point of view, given the API `findMyWay.off(method, path, constraints)`

- if the constraint is `=== undefined`, it is like calling `findMyWay.off(method, path)`
- if the constraint is falsy value (except `undefined`), it should match the route without constraints (as preference, I thing the empty object `{}` works as well)

Reply:
Sorry for the late response. @ivan-tymoshenko and @Eomm you have convinced me 👍 
what do you think about doing it like this?

- if the constraint is any falsy value (including `undefined`) it's like calling `findMyWay.off(method, path)` and all routes, regardless of the constraints, are deleted.
- if the constraint is an empty object only the route without constraints is deleted.

This is somewhat different in the handling of falsy values than the suggestion of @Eomm

Reply:
- if you want to delete only an unconstrained route, you should pass the empty object. This is the only way to do it.
- if you don't pass any parameter, except undefined, because it literally means nothing, all routes will be deleted.
- if you pass something other than an object, it should be your problem. We shouldn't try to guess what you want to do.

Reply:
So you suggest:

- findMyWay.off(method, path)  or findMyWay.off(method, path, undefined) &rarr; all routes will be deleted
- findMyWay.off(method, path, {}) &rarr; only the unconstrainted route will be deleted
- the third argument is neither undefined nor of type object &rarr; we will throw an error

Reply:
yes.